core/src/main/java/com/googlecode/psiprobe/tools/url/UrlParser.java

```diff
@@ -34,6 +34,13 @@
 
   /**
    * Instantiates a new url parser.
+   */
+  public UrlParser() {
```

Comment:
I reluctantly suggest you revert the recent changes and take it slower. Maybe add tests for one class at a time so it's not so much to review at once. They should be relatively independent anyway, I'd think.